ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -331,6 +374,72 @@ data ShelleyPartialLedgerConfig era = ShelleyPartialLedgerConfig {
     }
   deriving (Generic, NoThunks)
 
+instance Era era => FromCBOR (ShelleyPartialLedgerConfig era) where
+  fromCBOR =
+    ShelleyPartialLedgerConfig
+      <$> (ShelleyLedgerConfig
+        <$> fromCBOR
+        <*> (SL.Globals
+              -- Globals
+              --
+              -- Note: we can't serialise EpochInfo as it contains
+              -- lambdas, but that's ok because it's value is known
+              -- staticaly. It is always just `dummyEpochInfo` when inside
+              -- a ShelleyPartialLedgerConfi
+              dummyEpochInfo
+              <$> fromCBOR
+              <*> fromCBOR
+              <*> fromCBOR
+              <*> fromCBOR
+              <*> fromCBOR
+              <*> fromCBOR
+              <*> fromCBOR
+              <*> fromCBOR
+              <*> fromCBOR @ActiveSlotCoeff
```

Comment:
Are you applying a particular rule to determine which `fromCBOR`/`toCBOR` occurrences deserve type applications? It's not apparent to me.